### PR TITLE
ARTEMIS-3064 - ensure useTopologyForLoadBalancing is disabled when ha…

### PR DESCRIPTION
**Pull request by gtully**
…=false

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java [hunk 1]
```diff
@@ -69,6 +69,10 @@ public FederationConnection(Configuration configuration, String name, Federation
          }
       }
 
+      if (!config.isHA()) {
+         serverLocator.setUseTopologyForLoadBalancing(false);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java:73]: Should we maybe just like for others look to make this pick up from url configuration?

What do bridges do?

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java:73]: @gtully can you run the whole test suite and post results?

Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java:73]: interesting, https://issues.apache.org/jira/browse/ARTEMIS-347 introduces a cluster-connection-url in place of a connector-ref. The configuration is now a choice.
Maybe we can deprecate all of the server locator config for federation and just use a url? Or do we need to retain a choice? I think the server locator url parser is smart enough to cover all options w.r.t discovery groups etc.

Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java:73]: the tests look good. I think even is we provide an alternative url config, we will need this fix for the existing configuration.

I can raise an issue to bypass the connector-refs and have a connector-uri option where the rich url has all that is needed

Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederationConnection.java:73]: essentially a version of ARTEMIS-347 for federation

**Pull request closed without merging**